src/release/rollups.md

```diff
@@ -23,67 +23,99 @@ queue has been merged.
    deciding what to include, in particular on `rollup=maybe` and
    `rollup=iffy` PRs. We should try as much as possible to avoid risking
-   and hit regressions (bugs or perf).  Also consider that contributors
+   to hit regressions (bugs or perf).  Also consider that contributors
    often forget to tag things with rollup=never, when they should have
    done so, so when PRs are not explicitly tagged with rollup, be extra
-   careful.
+   careful. Also carefully consider the area of the compiler the PRs touch.
+   Two diagnostic PRs may actually conflict with each other, as they both 
+   change compiler output, which causes failures in each other's tests, 
+   when both of them are merged together in a rollup without causing git merge-conflicts.
+   In this case the older PR should be given the privilege to merge first 
+   and the newer PR should then be rebased as needed.
 
 2. Run the following command in the pull request thread:
 
     ```
     @bors r+ rollup=never p=5
     ````
-
+   where 5 is the number of PRs contained in your rollup.
 3. If the rollup fails, use the logs rust-log-analyzer
-   provides to bisect the failure to a specific PR and do
-   `@bors r-`. If the PR is running, you need to do `@bors r- retry`. Otherwise,
+   provides to bisect the failure to a specific PR and
+   `@bors r-` the PR. If the PR is running, you need to do `@bors r- retry`. Otherwise,
    your rollup succeeded. If it did, proceed to the next rollup (every now and
-   then let `rollup=never` and toolstate PRs progress).
-4. Recreate the rollup without the offending PR starting again from **1.**. There's a link in the rollup PR's body to automatically prefill the rollup UI with the existing PRs (minus any PRs that have been `r-`d)
+   then let `rollup=never/iffy` and toolstate PRs progress).
+4. Recreate the rollup without the offending PR starting again from **1.**. 
+   There's a link in the rollup PR's body to automatically prefill the rollup UI 
+   with the existing PRs (minus any PRs that have been `r-`d)
+   Try avoiding adding any additional PR to the current "batch" as this
+   unnecessarily increases the chance of test failures.
+   Rollups should not overlap, if a PR is already contained in a rollup that is not closed,
+   it should not be added to another different rollup at the same time.
+   If a rollup fails and you are not sure which PR caused the problem, 
+   you may bisect the rollup and split it up into two rollups until the offending PR becomes clear.
 
 ## Selecting Pull Requests
 
-The queue is sorted by rollup status. In general, a good rollup includes one or two `iffy` PRs (if available), a bunch of `maybe` (unmarked) PRs, and a large pile of `always` PRs. A rollup should never include `rollup=never` PRs.
+The queue is sorted by rollup status. In general, a good rollup contains a bunch of `maybe` (unmarked) PRs, and a large pile of `always` PRs. You can include one or two `iffy` PRs if you are confident that they will pass. 
```

Comment:
I'm not sure including "iffy" PRs is a good idea in rollups. I personally tend to never includ them.

Reply:
That's kinda the whole reason we have `"iffy"`. If you never include them, then they may as well be marked `"never"`.

Reply:
Fair point.

Reply:
I personally consider `iffy` PRs to be the whole point of rollups, tbh. Some PRs are clearly `rollup=never` if you are almost *certain* they will fail, but some of the `iffy` PRs are for the "it can fail, but we're not too sure" cases: it's great if it passes full CI, in which case we saved a bunch of time. It's also great if it fails full CI, in which case the failure time is amortized.